evap/results/templates/results_index.html

```diff
@@ -117,9 +117,10 @@ <h6 class="card-subtitle mb-1">{% trans 'Semester' %}</h6>
 
             // find which rows to show
             $('#results-table tr:not(.table-header)').each(function() {
+                var evaluationDegrees = $(this).find('.badge-degree').text();
                 var showRow =
                     (semesters.length == 0 || semesters.includes($(this).find('.semester-name').text())) &&
-                    (degrees.length == 0 || degrees.includes($(this).find('.badge-degree').text())) &&
+                    (degrees.length == 0 || degrees.some(s => evaluationDegrees.includes(s))) &&
```

Comment:
i'd rather inline this, or use the function as before, because right now the code is doing work for every line even if it's not required.